src/GraphQL.SystemTextJson/ObjectDictionaryConverter.cs

```diff
@@ -0,0 +1,223 @@
+                    case JsonValueKind.True:
+                    case JsonValueKind.False:
+                        resultValue = value.GetBoolean();
+                        break;
+                    case JsonValueKind.String:
+                        resultValue = value.GetString();
+                        break;
+                    case JsonValueKind.Null:
+                        // default value is null
+                        break;
+                    case JsonValueKind.Array:
+                        resultValue = ReadArray(value).ToList();
+                        break;
+                    default:
+                        throw new InvalidOperationException($"Unexpected value kind: {value.ValueKind}");
+                }
+
+                result[key] = resultValue;
+            }
+
+            if (!result.Any())
+                return null;
+
+            return result;
+        }
+
+        private IEnumerable<object> ReadArray(JsonElement value)
+        {
+            foreach (JsonElement item in value.EnumerateArray())
+            {
+                switch (item.ValueKind)
+                {
+                    case JsonValueKind.Object:
+                        yield return ReadDictionary(item);
+                        break;
+                    case JsonValueKind.Number:
+                        yield return ReadNumber(item);
+                        break;
+                    case JsonValueKind.True:
+                    case JsonValueKind.False:
+                        yield return item.GetBoolean();
+                        break;
+                    case JsonValueKind.String:
+                        yield return item.GetString();
+                        break;
+                    case JsonValueKind.Null:
+                        yield return null;
+                        break;
+                    case JsonValueKind.Array:
+                        yield return ReadArray(item).ToList();
+                        break;
+                    default:
+                        throw new InvalidOperationException($"Unexpected value kind: {item.ValueKind}");
+                }
+            }
+        }
+
+        private object ReadNumber(JsonElement value)
+        {
+            if (value.TryGetInt32(out var i))
+                return i;
+            else if (value.TryGetInt64(out var l))
+                return l;
+            //else if (value.TryGetUInt64(out var ui))
+            //    resultValue = ui;
+            else if (BigInteger.TryParse(value.GetRawText(), out var bi))
+                return bi;
+            else if (value.TryGetDouble(out var d))
+                return d;
+            else if (value.TryGetDecimal(out var dd))
+                return dd;
+
+            throw new NotImplementedException($"Unexpected Number value. Raw text was: {value.GetRawText()}");
+        }
+
+        private void WriteDictionary(Utf8JsonWriter writer, Dictionary<string, object> dictionary,
```

Comment:
I think you don't actually need these write methods.

Reply:
Yep, removed. Just pushed without.